datajoint/table.py

```diff
@@ -344,23 +348,49 @@ def _delete_cascade(self):
             try:
                 delete_count += self.delete_quick(get_count=True)
             except IntegrityError as error:
-                match = foregn_key_error_regexp.match(error.args[0])
-                assert match is not None, "foreign key parsing error"
+                try:
+                    # try to parse error for child info
+                    match = foreign_key_full_error_regexp.match(error.args[0]).groupdict()
+                    if "`.`" not in match['child']:  # if schema name missing, use self
+                        match['child'] = '{}.{}'.format(self.full_table_name.split(".")[0],
+                                                        match['child'])
+                    match['fk_attrs'] = [k.strip('`') for k in match['fk_attrs'].split(',')]
+                    match['pk_attrs'] = [k.strip('`') for k in match['pk_attrs'].split(',')]
+                except AttributeError:
+                    # additional query required due to truncated error, trying partial regex
```

Comment:
The implementation in this block will work in both cases of truncated and full error. Let's just keep this one implementation. The performance hit is minimal but the code will be simpler and more maintainable and testable.

Reply:
@dimitri-yatsenko True but I don't feel it makes sense to perform an entire additional query for every single instance of a foreign key on deletes when it can be avoided easily for cases where the error message has not been truncated. Specially considering that this optimization was all initiated based on performance issues with the prior implementation.

It is unfortunate that the drawback to the optimization of parsing the error messages comes with the MySQL 'near' error truncation policy but it seems like most relational database services perform something similar anyway.

If the concern is more towards 'readability' and how this might translate to maintenance and code coverage, I think I can refine this a bit more. Something to note as well is that this solution adds only 6 statements compared to the prior approach.

In any case, let me follow up with another commit to address this.

Reply:
@dimitri-yatsenko I have updated the PR with more refinements. Compared to the prior solution, there is now only an increase of 3 additional statements with these proposed changes. I have also removed the try/except block.